llvm-hs-pure/src/LLVM/IRBuilder/Instruction.hs

```diff
@@ -318,3 +318,11 @@ globalStringPtr str nm = do
   return $ C.GetElementPtr True
                            (C.GlobalReference (ptr ty) nm)
                            [(C.Int 32 0), (C.Int 32 0)]
+
+-- platform independant sizeof: a gep to the end of a nullptr and some bitcasting.
+sizeof :: LLVM.AST.Type -> C.Constant
```

Comment:
How about moving this to `LLVM.AST.Constant` and maybe reexport it from this module? It doesn’t seem to be particularly tied to the IRBuilder.

Reply:
Now that I think about this it may be most appropriate in LLVM.AST.Typed ?

Reply:
`Typed` is more for getting the type of something. `sizeof` goes the other way around. `LLVM.AST.Type` doesn’t even import `Constant` atm so it also doesn’t seem like a great home for this.